cocos2d/core/base-ui/CCWidgetManager.js

```diff
@@ -24,7 +24,14 @@
  THE SOFTWARE.
  ****************************************************************************/
 
-var Event = require('../CCNode').EventType;
+var Event;
+
+// Widget depends on WidgetManager
+// The main process needs the require widget
+// for cocos-creator/2d-tasks/issues/1894
+if (!CC_EDITOR || !Editor.isMainProcess) {
```

Comment:
// Support serializing widget in asset db, see cocos-creator/2d-tasks/issues/1894